### PR TITLE
Various style fixes for profile updates

### PR DESCRIPTION
**Pull request by uzi**
* Restyle follow/subscribe buttons
* Display more things by their displayNamePrefixed format
* Display names with canonical casing
* Other little cleanups

👓  @prashtx @nramadas @schwers

### CODE REVIEW

File: src/app/components/CommunityHeader/index.jsx [hunk 1]
```diff
@@ -67,18 +67,12 @@ const renderBannerRow = (subreddit, theme) => {
...
 
-const followIconClass = subscriber => {
-  return subscriber ? 'icon-check-circled lime' : 'icon-follow blue';
-};
-
 const renderSubscribeButton = subscriber => (
-  <button type='submit' className='CommunityHeader-text-row-blue CommunityHeader-no-outline'>
+  <button
+    type='submit'
+    className={ `CommunityHeader-subscribe${ subscriber ? 'r' : '' }-button` }
```
Comment (prashtx) [inline src/app/components/CommunityHeader/index.jsx]: nit: I think it's a little clearer to have a condition that selects between `CommunityHeader-subscribe-button` and `CommunityHeader-subscriber-button`, rather than selecting between `'r'` and `''`. It also avoid generating some garbage, since we'll just have two string literals.

Comment (uzi) [inline src/app/components/CommunityHeader/index.jsx]: fffffffffffine...

File: src/app/components/CommunityOverlayMenu/index.jsx [hunk 2]
```diff
@@ -38,8 +38,13 @@ const renderSubscriptions = (subscriptions, loading, theme) => (
...
           text={ (
             <span>
-              <span className='CommunityRow__rSlash'>r/</span>
-              { subreddit.displayName }
+              <span className='CommunityRow__rSlash'>
+                { subreddit.subredditType === 'user' ? 'u/' : 'r/' }
+              </span>
+              { subreddit.subredditType === 'user'
+                ? subreddit.displayName.substring(2)
+                : subreddit.displayName
```
Comment (prashtx) [inline src/app/components/CommunityOverlayMenu/index.jsx]: I think it's clearer here to use `subreddit.displayNamePrefixed` (which clearly indicates that there's some form of prefix in the string) and then pulling out the first two characters (`str.substring(0,2)`) and the remainder (`str.substring(2)`). Then we don't even need to condition on the subreddit type. Thoughts?

File: src/app/components/CommunityRow/index.js [hunk 3]
```diff
@@ -30,13 +30,22 @@ const renderIcon = (iconUrl, url, color, theme) => {
...
 
   return (
     <Anchor className='CommunityRow__details' href={ url }>
       <div className='CommunityRow__name'>
-        <span className='CommunityRow__rSlash'>r/</span>
-        { displayName }
+        <span className='CommunityRow__rSlash'>
+          { subredditType === 'user' ? 'u/' : 'r/' }
+        </span>
+        { subredditType === 'user' ? displayName.substring(2) : displayName }
```
Comment (prashtx) [inline src/app/components/CommunityRow/index.js]: Same thought as in CommunityOverlayMenu.

File: src/app/components/UserProfileHeader/index.jsx [hunk 4]
```diff
@@ -11,18 +11,9 @@ import SubredditSubscribeForm from 'app/components/SubredditSubscribeForm';
 
 const T = React.PropTypes;
 
-const followIconClass = follower => {
-  return follower ? 'icon-check-circled lime' : 'icon-follow blue';
-};
-
 const renderFollowButton = follower => (
-  <button type='submit' className='UserProfileHeader__text-row-blue UserProfileHeader__no-outline'>
+  <button type='submit' className={ `UserProfileHeader__follow${follower ? 'er' : '' }` }>
```
Comment (prashtx) [inline src/app/components/UserProfileHeader/index.jsx]: Same nit as in CommunityHeader above.

File: src/app/pages/UserActivity/index.jsx [hunk 5]
```diff
@@ -14,7 +14,7 @@ import { paramsToActiviesRequestId } from 'app/models/ActivitiesRequest';
 
 const mapStateToProps = createSelector(
   userAccountSelector,
-  (state, props) => state.accounts[props.urlParams.userName.toLowerCase()],
+  (state, props) => state.accounts[props.urlParams.userName.toLowerCase()] || {},
```
Comment (prashtx) [inline src/app/pages/UserActivity/index.jsx]: Do we use reference equality to determine if these selectors have changed value? If so, I think the `{}` will yield a different object after unrelated state changes. I think we can avoid that by returning `myUser || {}` in the combiner function, though it feels like passing an empty object will mask potential bugs where we should know that we don't yet have user data.

File: src/app/pages/UserProfile/index.jsx [hunk 6]
```diff
@@ -14,7 +14,7 @@ const GILD_URL_RE = /u\/.*\/gild$/;
 
 const mapStateToProps = createSelector(
   userAccountSelector,
-  (state, props) => state.accounts[props.urlParams.userName.toLowerCase()],
+  (state, props) => state.accounts[props.urlParams.userName.toLowerCase()] || {},
```
Comment (prashtx) [inline src/app/pages/UserProfile/index.jsx]: Same thought as in UserActivity.

**Comment by prashtx** [comment]
A couple of small thoughts and nits. Also, the build is failing on account of unused variables, I believe.

💅

File: src/app/pages/UserActivity/index.jsx (hunk 5 above)
Comment (uzi) [inline src/app/pages/UserActivity/index.jsx]: Fair -- fixed

File: src/app/pages/UserProfile/index.jsx (hunk 6 above)
Comment (uzi) [inline src/app/pages/UserProfile/index.jsx]: Fixing here, too

File: src/app/components/CommunityOverlayMenu/index.jsx (hunk 2 above)
Comment (uzi) [inline src/app/components/CommunityOverlayMenu/index.jsx]: Agreed -- except I think I'll split on `/`

**Comment by uzi** [comment]
💇 See if this is more to your liking

**Comment by prashtx** [comment]
:fish:
